bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetThingHandler.java

```diff
@@ -0,0 +1,205 @@
+                }
+                return;
+            }
+        }
+        logger.warn("No bridge associated, please assign a bridge in thing configuration. thing={}", thing.getUID());
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                "No bridge associated, please assign a bridge in thing configuration.");
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channel, Command command) {
+        logger.debug("handleCommand() (command={} - channel={})", command, channel);
+        OpenWebNetBridgeHandler handler = bridgeHandler;
+        if (handler != null) {
+            OpenGateway gw = handler.gateway;
+            if (gw != null && !gw.isConnected()) {
+                logger.warn("Gateway is NOT connected, setting thing={} to OFFLINE", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+                return;
+            }
+            if (command instanceof RefreshType) {
+                logger.debug("Refreshing channel {}", channel);
+                requestChannelState(channel);
+                // set a schedule to put device OFFLINE if no answer is received after THING_STATE_REQ_TIMEOUT
+                scheduler.schedule(() -> {
+                    // if state is still unknown after timer ends, set the thing OFFLINE
+                    if (thing.getStatus().equals(ThingStatus.UNKNOWN)) {
+                        logger.info("Thing state request timer expired, still unknown. Setting thing={} to OFFLINE",
+                                thing.getUID());
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "Could not get channel state");
+                    }
+                }, THING_STATE_REQ_TIMEOUT, TimeUnit.SECONDS);
+            } else {
+                handleChannelCommand(channel, command);
+            }
+        } else {
+            logger.info("Thing {} is not associated to any gateway, skipping command", getThing().getUID());
+        }
+    }
+
+    /**
+     * Handles a command for the specific channel for this thing.
+     * It must be implemented by each specific OpenWebNet category of device (WHO), based on channel
+     *
+     * @param channel specific ChannleUID
+     * @param command the Command to be executed
+     */
+    protected abstract void handleChannelCommand(ChannelUID channel, Command command);
+
+    /**
+     * Handle incoming message from OWN network directed to a device. It should be further implemented by each specific
+     * device handler.
+     *
+     * @param msg the message to handle
+     */
+    protected void handleMessage(BaseOpenMessage msg) {
+        ThingStatus ts = getThing().getStatus();
+        if (ThingStatus.ONLINE != ts && ThingStatus.REMOVING != ts && ThingStatus.REMOVED != ts) {
+            updateStatus(ThingStatus.ONLINE);
```

Comment:
I could have a message coming from the openwebnet network, received as callback by Bridge handler with `OpenWebNetBridgeHandler.onEventMessage`, and delivered to this ThingHandler *exactly* while the Thing is in REMOVING status.
With:
`ThingStatus.REMOVING != ts`
I avoid the risk to make the thing become wrongly ONLINE again (it's also a not allowed transition from `ThingStatus.REMOVING`).
Then the removing process will complete in `OpenWebNetThing.dispose()`, where the device id is unregistered from bridge handler so that new openwebnet messages will not be delivered to this disposed thing anymore.
It probably added this check because during testing this happened at least in few cases.

Reply:
You're right. This can happen. I didn't expect this behavior.